All_Relationships/6.true_relationship_preparation.py

```diff
@@ -0,0 +1,228 @@
+    odds_ratio, p_value = fisher_exact(c_table, alternative='greater')
+    odds.append(odds_ratio)
+    p_val.append(p_value)
+    
+    
+    total_disease = candidate_df[candidate_df["disease_id"] == disease]["sentence_count"].sum()
+    total_gene = candidate_df[candidate_df["gene_id"] == gene]["sentence_count"].sum()
+    expected.append((total_gene * total_disease)/float(total))
+
+candidate_df["nlog10_p_value"] = -np.log10(p_val)
+candidate_df["odds_ratio"] = odds
+candidate_df["expected_sen"] = expected
+candidate_df["lower_ci"] = lower_ci
+
+
+# In[ ]:
+
+candidate_df.sort_values("nlog10_p_value", ascending=False).head(20)
+
+
+# # Combine Sentence Marginal Probabilities
+
+# In this section we incorporate the marginal probabilites that are calculated from the bi-directional LSTM used in the [previous notebook](4.sentence-level-prediction.ipynb). For each sentence we grouped them by their disease-gene mention and report their marginal probabilites in different quantiles (0, 0.2, 0.4, 0.6, 0.8). Lastly we took the average of each sentence marginal to generate the "avg_marginal" column.
+
+# In[5]:
+
+candidate_df = pd.read_csv("disease_gene_summary_stats.csv")
+
+
+# In[6]:
+
+train_marginals_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/lstm_train_marginals.csv")
+dev_marginals_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/lstm_dev_marginals.csv")
+test_marginals_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/lstm_test_marginals.csv")
+
+
+# In[7]:
+
+train_sentences_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/train_candidates_sentences.csv")
+dev_sentences_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/dev_candidates_sentences.csv")
+test_sentences_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/test_candidates_sentences.csv")
+
+
+# In[8]:
+
+train_sentences_df["marginals"] = train_marginals_df["RNN_marginals"].values
+dev_sentences_df["marginals"] = dev_marginals_df["RNN_marginals"].values
+test_sentences_df["marginals"] = test_marginals_df["RNN_10_Marginals"].values
+
+
+# In[9]:
+
+candidate_marginals = (
+    train_sentences_df[["disease_id", "gene_id", "marginals"]]
+    .append(dev_sentences_df[["disease_id", "gene_id", "marginals"]])
+    .append(test_sentences_df[["disease_id", "gene_id", "marginals"]])
+    )
+
+
+# In[17]:
+
+quantile_list = [0,0.2,0.4,0.6,0.8]
+quantile_data = []
+avg_marginals = []
+group = candidate_marginals.groupby(["disease_id", "gene_id"])
+
+for i, cand in tqdm.tqdm(candidate_df[["disease_id", "gene_id"]].iterrows()):
+    dg_series = group.get_group((cand["disease_id"], cand["gene_id"]))
+    avg_marginals.append(dg_series["marginals"].mean())
+    quantile_data.append(map(lambda x: dg_series["marginals"].quantile(x), quantile_list))
+    if i == 10:
+        break
+print len(quantile_data[0])
+# Save the evidence into a dataframe
+candidate_df = pd.concat(
+    [
+        candidate_df.head(11),
```

Comment:
What is this `.head(11)`

Reply:
Removing that now. It was for testing the newly formatted code and I forgot to remove.